src/brod_client.erl

```diff
@@ -719,6 +734,34 @@ ensure_binary(ClientId) when is_atom(ClientId) ->
 ensure_binary(ClientId) when is_binary(ClientId) ->
   ClientId.
 
+maybe_start_metadata_sync(Config) ->
+  case config(sync_metadata, Config, false) of
+    true ->
+      erlang:send_after(sync_interval(Config), self(), sync);
+    false ->
+      undefined
+  end.
+
+sync_topics_metadata(Children, State) ->
+  lists:foldl(fun({Topic, _, _, _}, StateAcc) -> sync_topic_metadata(Topic, StateAcc) end, State, Children).
+
+sync_topic_metadata(Topic, State) ->
+  case do_get_metadata(Topic, State) of
+    {{ok, #{ topics := [ #{ partitions := Partitions } ] }}, NewState} ->
+      lists:foreach(fun (Partition) -> sync_topic_partition(Partition, Topic, State) end, Partitions),
+      NewState;
+    {{error, _}, NewState} ->
+      NewState
+  end.
+
+sync_topic_partition(#{ partition_index := Partition }, Topic, #state{ producers_sup = Sup, client_id = Client, config = Config}) ->
+  case brod_producers_sup:find_producer(Sup, Topic, Partition) of
+    {ok, _} ->
+      ok;
+    {error, {producer_not_found, _, _}} ->
+      brod_producer:start_link(Client, Topic, Partition, Config)
```

Comment:
should call brod_supervisor3:start_child to start the newly discovered producers?